src/metabase/api/dashboard.clj

```diff
@@ -336,24 +379,78 @@
+        new-mappings                   (for [{mappings :parameter_mappings, dashcard-id :id} dashcards
+                                             mapping                                         mappings
+                                             :when                                           (not (existing-mapping? dashcard-id mapping))]
+                                         (assoc mapping :dashcard-id dashcard-id))
+        ;; need to add the appropriate `:card-id` for all the new mappings we're going to check.
+        dashcard-id->card-id           (when (seq new-mappings)
+                                         (db/select-id->field :card_id DashboardCard
+                                           :dashboard_id dashboard-id
+                                           :id           [:in (set (map :dashcard-id new-mappings))]))
+        new-mappings (for [{:keys [dashcard-id], :as mapping} new-mappings]
+                       (assoc mapping :card-id (get dashcard-id->card-id dashcard-id)))]
+    (check-parameter-mapping-permissions new-mappings)))
+
+(def ^:private UpdatedDashboardCard
+  (su/with-api-error-message
+    {:id                                  (su/with-api-error-message su/IntGreaterThanOrEqualToZero
+                                            (tru "value must be a DashboardCard ID."))
+     (s/optional-key :sizeX)              (s/maybe su/IntGreaterThanZero)
+     (s/optional-key :sizeY)              (s/maybe su/IntGreaterThanZero)
+     (s/optional-key :row)                (s/maybe su/IntGreaterThanOrEqualToZero)
+     (s/optional-key :col)                (s/maybe su/IntGreaterThanOrEqualToZero)
+     (s/optional-key :parameter_mappings) (s/maybe [{:parameter_id su/NonBlankString
+                                                     :target       s/Any
+                                                     s/Keyword     s/Any}])
+     (s/optional-key :series)             (s/maybe [su/Map])
+     s/Keyword                            s/Any}
+    (tru "value must be a valid DashboardCard map.")))
+
 ;; TODO - we should use schema to validate the format of the Cards :D
 (api/defendpoint PUT "/:id/cards"
   "Update `Cards` on a Dashboard. Request body should have the form:
 
-    {:cards [{:id     ...
-              :sizeX  ...
-              :sizeY  ...
-              :row    ...
-              :col    ...
-              :series [{:id 123
-                        ...}]} ...]}"
+    {:cards [{:id                 ... ; DashboardCard ID
+              :sizeX              ...
+              :sizeY              ...
+              :row                ...
+              :col                ...
+              :parameter_mappings ...
+              :series             [{:id 123
+                                    ...}]}
+             ...]}"
   [id :as {{:keys [cards]} :body}]
+  {cards (su/non-empty [UpdatedDashboardCard])}
```

Comment:
Not specifically related to this change but I added a schema for validation and better dox to make it clearer what one is expected to pass in here